ui/apps/platform/src/Containers/ComplianceEnhanced/ScanConfigs/Wizard/ScanConfigWizardForm.tsx

```diff
@@ -68,7 +76,18 @@ function ScanConfigPage(): ReactElement {
         {
             name: SELECT_PROFILES,
             id: SELECT_PROFILES_ID,
-            component: <ProfileSelection />,
+            component: (
+                <ProfileSelection
+                    profiles={profiles || []}
```

Comment:
nit: I think there are three places where you could put this fallback to account for no profiles being available
- in the Service, so that anywhere in the app that request profiles from the API is guaranteed to get at least an empty array `[]` and never null or undefined
- here, where you have it, but that puts the onus on the caller to always require the calling function to pass in a fallback
- in the `<ProfileSelection>` component itself as a default prop

As a radically defensive (paranoid) programmer, I think the component should always have default props for something like this. I wouldn't objective to the service also returning a default of the correct type. 

In either case, I think either-or props are the least desirable, because they are hardest to maintain.

Reply:
I was planning on returning an empty array [from the service](https://github.com/stackrox/stackrox/pull/8847/files#diff-20fc5effabdf924484f3b7aa415cec1d138fbb66baaea85b3d2d0c37fd14c0f8R206) but the problem is the `useRestQuery` hook decides the type of the data returned (`data: ReturnType | undefined;`). I could set the profile default to an empty array but I'm still passing in the potentially "undefined" profiles regardless unfortunately.

Reply:
If you make the component accept `Profile[] | undefined` and then set a default for that prop of `= []`, would that work?